packages/docusaurus-theme-classic/src/theme/BlogLayout/index.tsx

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import clsx from 'clsx';
+
+import Layout from '@theme/Layout';
+import BlogSidebar from '@theme/BlogSidebar';
+import TOC from '@theme/TOC';
+
+import type {Props} from '@theme/BlogLayout';
+
+function BlogLayout(props: Props): JSX.Element {
+  const {sidebar, toc, children, ...layoutProps} = props;
+  const hasSidebar = sidebar && sidebar.items.length > 0;
+
+  return (
+    <Layout {...layoutProps}>
+      <div className="container margin-vert--lg">
+        <div className="row docusaurus-jc-center">
+          {hasSidebar && (
+            <aside className="col col--3">
+              <BlogSidebar sidebar={sidebar!} />
+            </aside>
+          )}
+          <main
+            className={clsx('col', {
+              'col--7': hasSidebar,
+              'col--9': !hasSidebar,
+            })}>
+            {children}
+          </main>
+          {toc && (
```

Comment:
In my opinion we should always render the TOC layout container (col--2) even if the toc is absent (blog post list) or the toc is empty (blog post with no headings)

This permits to avoid having the blog layout being offset as we navigate from the blog list of a blog post. It was the case before.

Eventually we could add a blog plugin option to disable the toc globally, similarly to the sidebar, in which case we could make the blog post content larger. If `options.blogSidebarCount === 0 && !options.toc` => `col--12`? (or maybe `col--10 col--offset-1` as 12 might be too big?

Reply:
> This permits to avoid having the blog layout being offset as we navigate from the blog list of a blog post. It was the case before.

However, then we would lose the correct central positioning, it is OK?


> 12 might be too big?

That's a lot, so I wouldn't use more col-9 for content area.

Reply:
> However, then we would lose the correct central positioning, it is OK?

To me, what matters most is that the real blog post content is centered, not really having perfect centering of the group "optional sidebar + content + optional toc". 

In this screenshot, the group is centered:

![image](https://user-images.githubusercontent.com/749374/126198905-bfb8fafc-71ad-45f9-a3f2-303a8702fe84.png)

In this one, the content is centered. Not perfectly though (it's not a big deal imho), but to me it looks better and will permit to have a consistent layout on all the blog pages:

![image](https://user-images.githubusercontent.com/749374/126199094-17c0f7cd-9f15-41c5-a497-609c0c9c85fd.png)